src/lib/pipeline/FetchUnit.cc

```diff
@@ -53,6 +53,15 @@ void FetchUnit::tick() {
       // Set prediction to recorded value during loop buffer filling
       if (macroOp[0]->isBranch()) {
         macroOp[0]->setBranchPrediction(loopBuffer_.front().prediction);
+        // Calling predict() in order to log the branch in the branch
+        // predictor.  However, we are reusing the prediction from the loop
+        // buffer so we do not use the return value from predict()
+        branchPredictor_.predict(macroOp[0]->getInstructionAddress(),
+                                 macroOp[0]->getBranchType(),
+                                 macroOp[0]->getKnownOffset());
+        assert(pred.isTaken == loopBuffer_.front().prediction.isTaken &&
+               "New prediction differs from loop buffer prediction");
```

Comment:
Nnot sure if I'm missing something obvious but where is `pred` defined?

Reply:
If it should be from the `predict()` call from just above this assert, then I think the comment should say "we only use the return value from predict() for the subsequent assert"